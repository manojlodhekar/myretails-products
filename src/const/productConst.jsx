export default {
    PRODUCT_API_URL: '/assets/data/item-data.json',
    PRODUCT_MAIN_PAGE_SUBHEADER: 'Product Details',
    START_INST: 'Please click button for Products details',
    HEADER_TEXT: 'MyRetails Products Warehouse',
    PRODUCT_HIGHLIGHT: 'Products Highlights',
    FIND_STORE: 'find in store',
    PRO_HELPFUL: 'Most helpful 4-5 star review',    
    CON_HELPFUL: 'Most helpful 1-2 star review',
    ERROR_MESSAGE: 'Oops..Something went Wrong, please try again.',
    TARGET_BIG_LOGO: '/assets/Target-Logo.png',
    TARGET_LOGO: '/assets/Color-Target-logo.png',
    ADD_TO_CART: '/assets/addToCart.png',
    PICK_IN_STORE: '/assets/pickInStore.png',
    SHOPING_BAG_ICON : 'https://payload.cargocollective.com/1/0/6826/4717503/Cart.png',//'/assets/shopping-bag.svg',
    CART_BTN_LABEL: 'ADD TO CART',
    PICK_BTN_LABEL: 'PICK UP IN STORE',
    ADD_TO_REG: 'ADD TO REGISTRY',
    ADD_TO_LIST: 'ADD TO LIST',
    SHARE: 'SHARE',
    MY_RETAILS_INFO: 'MyRetail is a rapidly growing company with HQ in Richmond, VA and over 200 stores across the east coast. The companyâ€™s annual revenue last year was $5 billion and growing at the rate of 10% year over year. myRetail sells general merchandise products, including a fast growing fresh grocery segment. The stores average 80,000 sq. ft. in size and carry around 20,000 products. myRetail wants to provide a multi-channel experience for its customers online.'
}
    