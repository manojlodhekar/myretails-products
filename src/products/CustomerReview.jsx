import React from 'react';
import { getStarRating} from '../utils/ProductUtils';

const CustomerReview = (props) =>{
    
    let displayDate = new Date(props.customerReview.datePosted).toDateString().split(' ');
    displayDate = `${displayDate[1]} ${displayDate[2]}, ${displayDate[3]}`;

    return(
        <div>
            <div className="starCls"> {getStarRating(props.customerReview.overallRating)} </div>
            <div className="customerReviewTitle">{props.customerReview.title}</div>
            <div className="returnPolicyCls">{props.customerReview.review}</div>
            <div className="returnPolicyCls"><span style={{color: '#0066cc'}}>{props.customerReview.screenName}</span> {displayDate}</div>
        </div>
    )
}
export default CustomerReview;