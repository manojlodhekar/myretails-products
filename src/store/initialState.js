export default {
    products: {
        selectedProductDetail: {},
        productDetailImages:{},
        productDetailDescription:{},
        productDetailCustomerReview:{},
        productDetailOffers:{},
        productDetailPromotions:[],
        productDetailReturnPolicy:{},
        productApiError: false
    }
}