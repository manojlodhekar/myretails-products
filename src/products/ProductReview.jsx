import React, { Component } from 'react';
import { Grid, Divider } from 'semantic-ui-react';
import CustomerReview from './CustomerReview';
import { getStarRating} from '../utils/ProductUtils';
import productConst from '../const/productConst';


class ProductReview extends Component {

    constructor(props){
        super(props);
        this.state = {
            showReviewFlag: false
        }
    }
    showHideReview = () =>{
        this.setState({
            showReviewFlag : !this.state.showReviewFlag
        })
    }
    render(){
        
        const {ProductReviewDetail} = this.props;
        return(
            <div >
                <div>
                    <span className="ratingTxtCls"> 
                        <span className="starCls"> {getStarRating(5)} </span> 
                        <span>Overall</span>  
                        <span className="viewReviewCls"
                        onClick={(e) =>this.showHideReview()}> View all {ProductReviewDetail.Reviews.length} Reviews </span> 
                    </span>
                </div>
                <div>
                    <Grid divided='vertically' className="reviewContainer">
                        
                        <Grid.Row columns={2}>
                        <Grid.Column>
                            <div>
                                <div className="reviewTitleCls">PRO</div>
                                <div className="reviewTextCls">{productConst.PRO_HELPFUL}</div>
                                <Divider style={{width:'125%'}}/>
                                <CustomerReview customerReview={ProductReviewDetail.Pro[0]} />
                            </div>
                        </Grid.Column>
                        <Grid.Column>
                            <div>
                            <div className="reviewTitleCls">CON</div>
                                <div className="reviewTextCls">{productConst.CON_HELPFUL}</div>
                                <Divider style={{width:'110%'}}/>
                            <CustomerReview customerReview={ProductReviewDetail.Con[0]} />
                            </div>
                        </Grid.Column>
                        </Grid.Row>
                    </Grid>
                        {
                            this.state.showReviewFlag ?
                            <Grid divided='vertically' className="reviewContainer">
                            <Grid.Row columns={1}>
                            <Grid.Column>
                                <div>
                                    <div className="reviewTitleCls">
                                        All other Reviews
                                        <span className="viewReviewCls"
                                         onClick={(e) => this.showHideReview()}> Hide Review </span>
                                    </div>
                                    <Divider style={{width:'100%'}}/>
                                    {
                                        ProductReviewDetail.Reviews.map((review,index) =>{
                                            return (
                                            <div key={index}>
                                                <CustomerReview customerReview={review} />
                                                <Divider style={{width:'100%'}}/>
                                            </div>
                                            )
                                        })
                                    }
                                    <span className="viewReviewCls"
                                         onClick={(e) => this.showHideReview()}> Hide Review </span>
                                </div>
                            </Grid.Column>
                            </Grid.Row> 
                        </Grid>
                            : null
                        }
                        
                </div>
            </div>
        )
    };
}
export default ProductReview;