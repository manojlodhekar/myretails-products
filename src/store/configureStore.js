import { createStore, applyMiddleware } from 'redux';
import rootReducer from '../reducers';
import thunk from 'redux-thunk';    
import initialState from './initialState'

export default function configureStore() {
    return (createStore(
        rootReducer,initialState,applyMiddleware(thunk)
    ));
}