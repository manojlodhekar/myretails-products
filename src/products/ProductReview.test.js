import React from 'react';
import ReactDOM from 'react-dom';
import ProductReview from './ProductReview';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import initialState from '../store/initialState'
import CustomerReview from './CustomerReview';
import productConst from '../const/productConst';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let productReview;
const mockStore = configureStore();
const store = mockStore(initialState);
const reviewDetails = { "CustomerReview": [
  {
   "Con": [
    {
     "datePosted": "Mon Mar 11 13:13:55 UTC 2013",
     "title": "Very unhappy"
    }
   ],
   "Pro": [
    {
     "datePosted": "Thu Apr 18 19:42:19 UTC 2013",
     "title": "Fantastic Blender"
    }
   ],
   "Reviews": [
    {
     "city": "NYC",
     "totalVotes": "52"
    }]
  }]};

beforeEach(()=>{
  productReview = mount(<ProductReview ProductReviewDetail={reviewDetails}>
                                  <CustomerReview customerReview={reviewDetails.CustomerReview[0]} />
                        </ProductReview>
);

})
describe('ProductReview is defined', () => {

  it('should render correctly', () => {
    expect(productReview).toHaveLength(1);
  });

  it('should render view all review component', () => {
    productReview = shallow(<ProductReview ProductReviewDetail={reviewDetails.CustomerReview[0]} />);
    expect(productReview.find('.viewReviewCls').first().text()).toEqual(' View all 1 Reviews ');
  });
  it('should render curent PRO of component', () => {
    productReview = shallow(<ProductReview ProductReviewDetail={reviewDetails.CustomerReview[0]} />);
    expect(productReview.find('.reviewTitleCls').first().text()).toEqual('PRO');
  });
  it('should render curent CON of component', () => {
    productReview = shallow(<ProductReview ProductReviewDetail={reviewDetails.CustomerReview[0]} />);
    expect(productReview.find('.reviewTitleCls').last().text()).toEqual('CON');
  });

  it('should render curent PRO_HELPFUL of component', () => {
    productReview = shallow(<ProductReview ProductReviewDetail={reviewDetails.CustomerReview[0]} />);
    expect(productReview.find('.reviewTextCls').first().text()).toEqual(productConst.PRO_HELPFUL);
  });
  it('should render curent CON_HELPFUL of component', () => {
    productReview = shallow(<ProductReview ProductReviewDetail={reviewDetails.CustomerReview[0]} />);
    expect(productReview.find('.reviewTextCls').last().text()).toEqual(productConst.CON_HELPFUL);
  });
  it('should render inner component', () => {
    expect(productReview).toMatchSnapshot();
  });

 });
