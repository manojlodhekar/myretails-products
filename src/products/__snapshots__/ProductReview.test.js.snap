// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReview is defined should render inner component 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <ProductReview
    ProductReviewDetail={
        Object {
            "CustomerReview": Array [
              Object {
                "Con": Array [
                  Object {
                    "datePosted": "Mon Mar 11 13:13:55 UTC 2013",
                    "title": "Very unhappy",
                  },
                ],
                "Pro": Array [
                  Object {
                    "datePosted": "Thu Apr 18 19:42:19 UTC 2013",
                    "title": "Fantastic Blender",
                  },
                ],
                "Reviews": Array [
                  Object {
                    "city": "NYC",
                    "totalVotes": "52",
                  },
                ],
              },
            ],
          }
    }
>
    <CustomerReview
        customerReview={
            Object {
                "Con": Array [
                  Object {
                    "datePosted": "Mon Mar 11 13:13:55 UTC 2013",
                    "title": "Very unhappy",
                  },
                ],
                "Pro": Array [
                  Object {
                    "datePosted": "Thu Apr 18 19:42:19 UTC 2013",
                    "title": "Fantastic Blender",
                  },
                ],
                "Reviews": Array [
                  Object {
                    "city": "NYC",
                    "totalVotes": "52",
                  },
                ],
              }
        }
    />
</ProductReview>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
